yeonjae/programmers-2/다음큰숫자.js

```diff
@@ -0,0 +1,12 @@
+function solution(n) {
+  const checkOneNum = n.toString(2).match(/1/g).length;
+
+  while (n++) {
+    if (n.toString(2).match(/1/g).length === checkOneNum) {
+      return n;
+    }
+  }
+}
```

Comment:
```suggestion
function solution(n) {
  const targetOneNumCount = n.toString(2).match(/1/g).length;
  let decimalNum = n;

  while (decimalNum++) {
    const nextOneNumCount = decimalNum.toString(2).match(/1/g).length;
    
    if (nextOneNumCount === targetOneNumCount) {
      return decimalNum;
    }
  }
}
```
- 깔끔한 풇이잘봤습니다! 🙇‍♂️
- n을 바로 이용하기보다 특정 변수로 뺴보는건 어떨까요! 사이드 이펙트를 방지해봐요~
- while문 내부의 `n.toString().match().length`도 변수로서 빼서 활용하면 좀 더 가독성 높은 코드 작성이 가능 할 것 같습니다